.github/workflows/root-ci-config/build_root.py

```diff
@@ -379,24 +409,131 @@ def create_binaries(buildtype):
 
 
 @github_log_group("Rebase")
-def rebase(base_ref, head_ref) -> None:
-    head_ref_src, _, head_ref_dst = head_ref.partition(":")
-    head_ref_dst = head_ref_dst or "__tmp"
+def rebase(directory: str, repository:str, base_ref: str, head_ref: str, head_sha: str) -> None:
     # rebase fails unless user.email and user.name is set
     result = subprocess_with_log(f"""
-        cd '{WORKDIR}/src'
+        cd '{WORKDIR}/{directory}'
 
         git config user.email "rootci@root.cern"
         git config user.name 'ROOT Continous Integration'
 
-        git fetch origin {head_ref_src}:{head_ref_dst}
-        git checkout {head_ref_dst}
+        git fetch {repository} {head_sha}:{head_ref}
+        git checkout {head_ref}
         git rebase {base_ref}
     """)
 
     if result != 0:
         die(result, "Rebase failed")
 
+def get_stdout_subprocess(command: str, error_message: str) -> str:
+  """
+  get_stdout_subprocess
+  execute and log a command.
+  capture the stdout, strip white space and return it
+  die in case of failed execution unless the error_message is empty.
+  """
+  result  = subprocess_with_capture(command)
+  if result.returncode != 0:
+    if error_message != "":
+      die(result, error_message)
+    else:
+      print("\033[90m", end='')
+      print(result.stdout)
+      print(result.stderr)
+      print("\033[0m", end='')
+      return ""
```

Comment:
I'm curious about this policy, if the error message is empty we don't kill the process. But still the returncode tells us there was a problem, so why don't kill it either way?

Reply:
The case 'error_message is empty' is for when the caller wants to handle the error case.  In pattern like:
```
if (cmd_one fails)
   try something else
entity_does_not_exist = ( cmd_two_fails == true )
```